### PR TITLE
FWF-3460:[Feature]- Edit Page New Design & Universal scrolling

### PR DESCRIPTION
**Pull request by Josephalexantony-aot**
# Issue Tracking

JIRA: https://aottech.atlassian.net/browse/FWF-3460
https://aottech.atlassian.net/browse/FWF-3647
Issue Type: BUG/ FEATURE

# Changes:

Create | Edit UI, Listing page UI (scrolling)

# Screenshots (if applicable)
https://jam.dev/c/78d6b155-86b3-47c1-8b28-8c472a0b69ed 
image
image

### CODE REVIEW

File: forms-flow-web/src/components/Form/Item/Edit.js [hunk 1]
```diff
@@ -449,377 +249,264 @@ const Edit = React.memo(() => {
...
+                        onClick={saveLayout}
+                        dataTestid={"save-form-layout"}
+                        ariaLabel={"Save Form Layout"}
+                      />
+                      <CustomButton
+                        variant="secondary"
+                        size="md"
+                        label={<Translation>{(t) => t("Discard Changes")}</Translation>}
+                        onClick={discardChanges}
+                        dataTestid={"cancelBtndataTestid"}
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Form/Item/Edit.js:397]: please follow kebab casing while giving datattestid

File: forms-flow-web/src/components/Form/Item/Edit.js [hunk 2]
```diff
@@ -449,377 +249,264 @@ const Edit = React.memo(() => {
...
+                        ariaLabel={"cancelBtnariaLabel"}
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Form/Item/Edit.js:398]: give a meaningfull name for aria label

File: forms-flow-web/src/components/Form/Item/Edit.js [hunk 3]
```diff
@@ -449,377 +249,264 @@ const Edit = React.memo(() => {
...
+                      />
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Form/Item/Edit.js:472]: please follow kebab casing while giving datattestid
give a meaningfull name for aria label

File: forms-flow-web/src/components/Form/Item/Edit.js (hunk 2 above)
Comment (arun-s-aot) [inline forms-flow-web/src/components/Form/Item/Edit.js:398]: - please follow kebab casing while giving datatestid
- give a meaningful name for aria label
- give translation tag to the text (ex : {<Translation>{(t) => t("Discard Changes")}</Translation>} )

File: forms-flow-web/src/components/Form/Item/Edit.js [hunk 4]
```diff
@@ -449,377 +249,264 @@ const Edit = React.memo(() => {
...
-                        onChange={(event) => handleChange("display", event)}
+                    <div>
+                      <CustomButton
+                        variant="primary"
+                        size="md"
+                        className="mx-2"
+                        label={<Translation>{(t) => t("Save Layout")}</Translation>}
+                        onClick={saveLayout}
+                        dataTestid={"save-form-layout"}
+                        ariaLabel={"Save Form Layout"}
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Form/Item/Edit.js:390]: give translation tag to the arialabel text (ex : {{(t) => t("Discard Changes")}} )

File: forms-flow-web/src/components/Form/Item/Edit.js [hunk 5]
```diff
@@ -449,377 +249,264 @@ const Edit = React.memo(() => {
...
+                        />
+                        <CustomButton
+                          variant="secondary"
+                          size="md"
+                          className="mx-2"
+                          icon={<PreviewIcon />}
+                          label={<Translation>{(t) => t("Preview")}</Translation>}
+                          onClick={handlePreview}
+                          dataTestid={"cancelBtndataTestid"}
+                          ariaLabel={"cancelBtnariaLabel"}
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Form/Item/Edit.js:378]: please follow kebab casing while giving datatestid
give a meaningful name for aria label
give translation tag to the arialabeltext (ex : {{(t) => t("Discard Changes")}} )

File: forms-flow-web/src/components/Form/Item/Edit.js [hunk 6]
```diff
@@ -449,377 +249,264 @@ const Edit = React.memo(() => {
...
+                      <div className="mx-2 builder-header-text">Layout</div>
+                      <div>
+                        <CustomButton
+                          variant="secondary"
+                          size="md"
+                          icon={<HistoryIcon />}
+                          label={<Translation>{(t) => t("History")}</Translation>}
+                          onClick={handleHistory}
+                          dataTestid={"cancelBtndataTestid"}
+                          ariaLabel={"cancelBtnariaLabel"}
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Form/Item/Edit.js:368]: please follow kebab casing while giving datatestid
give a meaningful name for aria label
give translation tag to the arialabeltext (ex : {{(t) => t("Discard Changes")}} )

File: forms-flow-web/src/components/Form/Item/Edit.js [hunk 7]
```diff
@@ -449,377 +249,264 @@ const Edit = React.memo(() => {
...
+                    dataTestid={"cancelBtndataTestid"}
+                    ariaLabel={"cancelBtnariaLabel"}
+                  />
+                  <CustomButton
+                    variant="light"
+                    size="md"
+                    label={<Translation>{(t) => t(publisText)}</Translation>}
+                    onClick={handlePublish}
+                    dataTestid={"cancelBtndataTestid"}
+                    ariaLabel={"cancelBtnariaLabel"}
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Form/Item/Edit.js:347]: - please follow kebab casing while giving datatestid and give relevant naming
- give a meaningful name for aria label
- give translation tag to the arialabeltext (ex : {{(t) => t("Discard Changes")}} )

File: forms-flow-web/src/components/Form/Item/Edit.js [hunk 8]
```diff
@@ -449,377 +249,264 @@ const Edit = React.memo(() => {
...
+                  />
+                  <CustomButton
+                    variant="dark"
+                    size="md"
+                    className="mx-2"
+                    label={<Translation>{(t) => t("Actions")}</Translation>}
+                    onClick={editorActions}
+                    dataTestid={"cancelBtndataTestid"}
+                    ariaLabel={"cancelBtnariaLabel"}
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Form/Item/Edit.js:339]: - please follow kebab casing while giving datatestid and give relevant naming
- give a meaningful name for aria label
- give translation tag to the arialabeltext (ex : {{(t) => t("Discard Changes")}} )

File: forms-flow-web/src/components/Form/Item/Edit.js [hunk 9]
```diff
@@ -449,377 +249,264 @@ const Edit = React.memo(() => {
...
-                <div className="bg-white">
-                  <RichText data-testid="form-edit-description" value={formDescription} onChange={setFormDescription} />
+                <div>
+                  <CustomButton
+                    variant="dark"
+                    size="md"
+                    label={<Translation>{(t) => t("Settings")}</Translation>}
+                    onClick={editorSettings}
+                    dataTestid={"cancelBtndataTestid"}
+                    ariaLabel={"cancelBtnariaLabel"}
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Form/Item/Edit.js:330]: - please follow kebab casing while giving datatestid and give relevant naming
- give a meaningful name for aria label
- give translation tag to the arialabeltext (ex : {{(t) => t("Discard Changes")}} )

File: forms-flow-web/src/components/Form/Item/Edit.js [hunk 10]
```diff
@@ -449,377 +249,264 @@ const Edit = React.memo(() => {
...
-                    id="title"
-                    placeholder={t("Enter the form title")}
-                    value={form.title || ""}
-                    onChange={(event) => handleChange("title", event)}
-                  />
+          <Errors errors={errors} />
+
+          <Card className="editor-header">
+            <Card.Body>
+              <div className="d-flex justify-content-between align-items-center" style={{ width: "100%" }}>
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Form/Item/Edit.js:308]: avoid using inline style if possible

File: forms-flow-web/src/components/Form/Item/Edit.js [hunk 11]
```diff
@@ -68,191 +61,52 @@ const reducer = (form, { type, value }) => {
...
-    if (
-      newForm &&
-      (form._id !== newForm._id || form.modified !== newForm.modified)
-    ) {
-      dispatchFormAction({ type: "replaceForm", value: newForm });
+  //for save form 
+  const [formSubmitted, setFormSubmitted] = useState(false);
+  const formAccess = useSelector((state) => state.user?.formAccess || []);
+  const submissionAccess = useSelector((state) => state.user?.submissionAccess || []);
+  const prviousData = useSelector((state) => state.process?.formPreviousData);
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Form/Item/Edit.js:85]: Change variable name **prviousData** to **previousData** everywhere

Comment (Josephalexantony-aot) [reply forms-flow-web/src/components/Form/Item/Edit.js:85]: The issue came from the previous code.

**Review by arun-s-aot — APPROVED** [review]

**Review by Ajay-aot — APPROVED** [review]

**Review by fahad-aot — APPROVED** [review]

**Review by arun-s-aot — APPROVED** [review]

**Pull request merged**